arwen/contexts/metering_test.go

```diff
@@ -279,80 +282,212 @@ func TestMeteringContext_GasUsed_NoStacking(t *testing.T) {
 	require.Equal(t, uint64(1401), gasUsedByContract)
 }
 
-func TestMeteringContext_GasUsed_StackOneLevel(t *testing.T) {
-	t.Skip()
+func setUpStackOneLevel(t *testing.T) (*meteringContext, *outputContext, *contextmock.RuntimeContextMock, *vmcommon.ContractCallInput, uint64, *vmcommon.ContractCallInput) {
 	t.Parallel()
 
 	mockRuntime := &contextmock.RuntimeContextMock{}
+
 	host := &contextmock.VMHostMock{
 		RuntimeContext: mockRuntime,
 	}
 
+	output, _ := NewOutputContext(host)
+	host.OutputContext = output
+
 	contractSize := uint64(1000)
 	contract := make([]byte, contractSize)
+
 	mockRuntime.SCCodeSize = contractSize
 	mockRuntime.SCAddress = []byte("parent")
 
 	mockRuntime.SetPointsUsed(0)
-	parentInput := &vmcommon.ContractCallInput{VMInput: vmcommon.VMInput{}}
+	parentInput := &vmcommon.ContractCallInput{
+		VMInput: vmcommon.VMInput{},
+	}
 	mockRuntime.SetVMInput(&parentInput.VMInput)
 
 	metering, _ := NewMeteringContext(host, config.MakeGasMapForTests(), uint64(15000))
+	host.MeteringContext = metering
+	zeroCodeCosts(metering)
 
 	parentInput.GasProvided = 4000
 	metering.InitStateFromContractCallInput(&parentInput.VMInput)
-	require.Equal(t, uint64(4000), metering.initialGasProvided)
+	require.Equal(t, 4000, int(metering.initialGasProvided))
 
 	_ = metering.DeductInitialGasForExecution(contract)
-	require.Equal(t, uint64(2999), metering.GasLeft())
+
+	require.Equal(t, 3000, int(metering.GasLeft()))
 
 	metering.UseGas(400)
-	require.Equal(t, uint64(2599), metering.GasLeft())
+	require.Equal(t, 2600, int(metering.GasLeft()))
 
-	gasUsedByContract := metering.GasSpentByContract()
-	require.Equal(t, uint64(1401), gasUsedByContract)
+	gasSpentByContract := metering.GasSpentByContract()
+	require.Equal(t, 1400, int(gasSpentByContract))
 
-	// simulate executing another contract on top of the parent
 	childInput := &vmcommon.ContractCallInput{VMInput: vmcommon.VMInput{}}
 	childInput.GasProvided = 500
+	childInput.CallerAddr = parentInput.RecipientAddr
 
 	metering.UseGas(childInput.GasProvided)
 	parentPointsBeforeStacking := mockRuntime.GetPointsUsed()
+	require.Equal(t, 900, int(parentPointsBeforeStacking))
+	require.Equal(t, 2100, int(metering.GasLeft()))
 
-	// child execution begins
 	mockRuntime.SCAddress = []byte("child")
 	mockRuntime.SetPointsUsed(0)
 	mockRuntime.SetVMInput(&childInput.VMInput)
 	metering.PushState()
 	metering.InitStateFromContractCallInput(&childInput.VMInput)
-	require.Equal(t, uint64(500), metering.initialGasProvided)
+	require.Equal(t, 500, int(metering.initialGasProvided))
 
+	return metering, output, mockRuntime, parentInput, parentPointsBeforeStacking, childInput
+}
+
+func TestMeteringContext_GasUsed_StackOneLevel(t *testing.T) {
+	metering, output, mockRuntime, parentInput, parentPointsBeforeStacking, _ := setUpStackOneLevel(t)
```

Comment:
Consider instantiating `parentInput` here, then pass it to `setUpStackOneLevel()`.